indico/modules/events/contributions/templates/display/contribution_display.html

```diff
@@ -141,7 +141,7 @@
 {%- endblock %}
 
 {% block content -%}
-    {%- if not published -%}
+    {%- if not published and contribution.event.can_manage(session.user) -%}
```

Comment:
That'll show it for a manager even if they are associated with it.. I'd add the association check here as well, i.e. `not published and manager and not associated`